apps/molgenis-components/src/components/layout/CookieWall.vue

```diff
@@ -0,0 +1,115 @@
+<script setup>
+import { computed, ref } from "vue";
+import { useCookies } from "@vueuse/integrations/useCookies";
+import LayoutModal from "./LayoutModal.vue";
+import ButtonAction from "../forms/ButtonAction.vue";
+import ButtonAlt from "../forms/ButtonAlt.vue";
+import GTag from "./GTag.vue";
+const cookies = useCookies(["mg_allow_analytics"]);
+const props = defineProps({
+  title: {
+    type: String,
+    default: "Would you like a cookie ?",
+  },
+  acceptLabel: {
+    type: String,
+    default: "Accept",
+  },
+  rejectLabel: {
+    type: String,
+    default: "No thanks",
+  },
+  analyticsId: {
+    type: String,
+  },
+});
+
+const emit = defineEmits(["acceptCookie"]);
+
+let isDeclined = ref(false);
+
+const show = computed(() => {
+  return cookies.get("mg_allow_analytics") === undefined;
+});
+
+const cookieOptions = {
+  path: "/",
+  maxAge: 1704085200, // one year,
+  httpOnly: false,
+};
+
+function handleAccept() {
+  cookies.set("mg_allow_analytics", true, cookieOptions);
+  emit("acceptCookie", true);
+  window.location.reload();
+}
+
+function handleDecline() {
+  cookies.set("mg_allow_analytics", false, cookieOptions);
+  emit("acceptCookie", false);
+}
+
+function handleClose() {
+  isDeclined.value = true;
+  emit("acceptCookie", undefined);
+}
+
+const isAnalyticsEnabled = computed(() => {
+  return cookies.get("mg_allow_analytics") === true;
+});
+</script>
+
+<template>
+  <LayoutModal :title="title" @close="handleClose" :show="show && !isDeclined">
+    <template v-slot:body>
+      <slot>
+        <p>
+          We use cookies and similar technologies to enhance your browsing
```

Comment:
usually there is a link provided to the privacy statement. I understand this is template code, but is there a way for a user to add this at runtime? 

It would be also nice to at least refer to the privacy policy like: 

For more details see the privacy policy.

Reply:
yes , this pr is meant as tech setup/test , should be part of future PR when feature is turned on in prod system

Reply:
After PR has been testen , i now added some code to update the content ( allow admin to set string or html)